examples/solvers/chambolle_pock_denoising_complex.py

```diff
@@ -0,0 +1,83 @@
+"""Total variation denoising of complex data using the Chambolle-Pock solver.
+
+Solves the optimization problem
+
+    min_{x}  1/2 ||x - g||_2^2 + lam || |grad(x)| ||_1
+
+Where ``grad`` the spatial gradient and ``g`` is given noisy data.
+
+For further details and a description of the solution method used, see
+:ref:`chambolle_pock` in the ODL documentation.
+"""
+
+import numpy as np
+import scipy
+import odl
+
+# Read test image: use only every second pixel, convert integer to float,
+# and rotate to get the image upright
+dtype = np.complex64
+image = np.rot90(scipy.misc.ascent()[::1, ::1], 3).astype(dtype)
+image = image + 1j*image.T
+shape = image.shape
+
+# Rescale max to 1
+image /= image.real.max()
+
+# Discretized spaces
+space = odl.uniform_discr([0, 0], shape, shape, dtype=dtype)
+
+# Original image
+orig = space.element(image)
+
+# Add noise
+image += (np.random.normal(0, 0.05, shape) +
```

Comment:
If you instead use

```python
noisy = image + 0.05 * odl.phantom.white_noise(image.space)
```

the noise is implicitly complex valued if the space is so.